### PR TITLE
Python dependency replaced, build script added, and typescript removed

### PR DESCRIPTION
**Pull request by rakusan2**
This is the same as #21 only with typescript removed

### CODE REVIEW

**Comment by ErikBoesen** [comment]
Once I have a testing environment I'll merge this. Thank you so much!

**Comment by ErikBoesen** [comment]
Sorry I'm just now getting to this. What is the purpose of the `tryIPC.js` file?

**Comment by ErikBoesen** [comment]
Whoops

**Comment by rakusan2** [comment]
The `tryIPC.js` is to tell the rest of the program that it is running through electron
Since the connect popup does not work in `pynetworktables2js`

**Comment by ErikBoesen** [comment]
Okay, this looks good. One thing: Could you add an option to run the dashboard without connecting to a robot? For testing purposes?

**Comment by ErikBoesen** [comment]
Also, "Connecting" isn't centered in the button.

**Comment by rakusan2** [comment]
Currently to dismiss the popup all you need to do is double press Esc

File: README.md [hunk 1]
```diff
@@ -42,8 +36,15 @@ In order to run the camera, you must start an `mjpg-streamer` server on the Robo
         python3 installer.py download-opkg mjpg-streamer
         python3 installer.py install-opkg mjpg-streamer
 
-3. Update `style.css` to use the IP of your live camera feed. Usually this is something like `roborio-XXXX-frc.local:5800/?action=stream`, where `XXXX` is your team's number.
+3. Update `style.css` to use the IP of your live camera feed. Usually this is something like `roborio-XXXX-frc.local:5800/?action=stream`, where `XXXX` is your team's number. The port may vary.
+
 
+## Building
+1. Get `npm`
+2. run `npm i` to install all of the depedencies
```
Comment (ErikBoesen) [inline README.md]: Capitalize run and fix spelling of "dependencies."

File: css/style.css [hunk 2]
```diff
@@ -110,47 +122,108 @@ svg * {
...
     fill: white;
 }
+
 #gyro {
     width: 175px;
     height: 175px;
     margin-left: 30px;
     margin-top: 15px;
 }
+
```
Comment (ErikBoesen) [inline css/style.css]: Get rid of these newlines.

File: index.html [hunk 3]
```diff
@@ -19,30 +19,30 @@
...
         <button id="get">Get</button>
     </div>
     <!-- The main body of the dashboard, by default contains the gyro monitor and the robot diagram -->
     <article>
         <!-- Gyroscope -->
         <svg id="gyro">
-            <circle cx="85" cy="85" r="83" />
-            <circle cx="85" cy="85" r="50" />
-            <rect id="gyro-arm" x="80" y="5" width="10" height="80" />
+            <circle cx="85" cy="85" r="83"></circle>
```
Comment (ErikBoesen) [inline index.html]: Since when do `<circle>`s need closing tags?

File: index.html [hunk 4]
```diff
@@ -65,7 +65,14 @@
             <input id="arm-position" type="range" min="0" max="1200" value="770">
         </div>
     </aside>
-    <script src="/networktables/networktables.js"></script>
+    <div class="greybox">
```
Comment (ErikBoesen) [inline index.html]: `.greybox` is a bit of an odd classname.

File: main.js [hunk 5]
```diff
@@ -1,95 +1,116 @@
...
+    console.log("Application quit.");
 });
 
-app.on('activate', function() {
-	// On OS X it's common to re-create a window in the app when the
-	// dock icon is clicked and there are no other windows open.
-	if (mainWindow === null) {
-		createWindow();
-	}
+app.on("activate", function () {
```
Comment (ErikBoesen) [inline main.js]: Use single quotes

File: package.json [hunk 6]
```diff
@@ -17,6 +19,14 @@
...
-    "electron": "^1.2.0"
+    "electron": "^1.2.0",
+    "electron-builder": "~15.2.0"
+  },
+  "dependencies": {
+    "wpilib-nt-client": "^1.1.3"
+  },
+  "build": {
+    "appId": "SmartDashboardJS",
+    "productName": "SmartDashboardJS"
```
Comment (ErikBoesen) [inline package.json]: SmartDashboardJS?? It's called FRCDashboard.

**Comment by ErikBoesen** [comment]
Oh really? Alright.

File: index.html (hunk 3 above)
Comment (rakusan2) [inline index.html]: That was the formater in VSCode that I use that did that

**Comment by ErikBoesen** [comment]
Also, the tuning panel transitions from blurred to unblurred every time it's opened.

**Comment by rakusan2** [comment]
You should know that I am still new to HTML and CSS, and that transition was meant to be for when the popup is closed but I never got it to work and forgot to remove it

File: index.html (hunk 4 above)
Comment (rakusan2) [inline index.html]: When the popup shows itself two things happen:
- The background is slightly grayed out
- Anything behind the box graying out the background is blured

**Comment by rakusan2** [comment]
I have made the little changes that you requested

**Pull request merged**